packages/builder/src/components/automation/AutomationBuilder/FlowChart/FlowItemHeader.svelte

```diff
@@ -1,27 +1,39 @@
 <script>
-  import { automationStore } from "builderStore"
-  import { Icon, Body, Detail, StatusLight } from "@budibase/bbui"
+  import { automationStore, selectedAutomation } from "builderStore"
+  import { Icon, Body, StatusLight, AbsTooltip } from "@budibase/bbui"
   import { externalActions } from "./ExternalActions"
   import { createEventDispatcher } from "svelte"
+  import { Features } from "constants/backend/automations"
 
   export let block
   export let open
   export let showTestStatus = false
   export let testResult
   export let isTrigger
   export let idx
+  export let addLooping
+  export let deleteStep
+
+  let validRegex = /^[A-Za-z0-9_\s]+$/
+  let typing = false
 
   const dispatch = createEventDispatcher()
 
+  $: stepNames = $selectedAutomation.definition.stepNames
+  $: automationName = stepNames[block.id] || block.name || ""
+  $: automationNameError = getAutomationNameError(automationName)
+  $: status = updateStatus(testResult, isTrigger)
+  $: isTrigger = isTrigger || block.type === "TRIGGER"
   $: {
     if (!testResult) {
       testResult = $automationStore.testResults?.steps?.filter(step =>
         block.id ? step.id === block.id : step.stepId === block.stepId
       )?.[0]
     }
   }
-  $: isTrigger = isTrigger || block.type === "TRIGGER"
-  $: status = updateStatus(testResult, isTrigger)
+  $: loopBlock = $selectedAutomation?.definition.steps.find(
```

Comment:
`selectedAutomation` has optional chaining here, but up above we're referencing a nested property without it. I think this should be consistent - it either can be null (in which case we need to update the `stepNames` declaration above) or it can't be null (in which case we don't need optional chaining).

I reckon just try deleting the selected automation and see if it throws an error or not - then we'll know whether this gets rendered one final time when the selected automation is null.